example-app/app/Models/Flight.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+use App\Exceptions\{
+    ArrivalCityNotAssociatedException,
+    DepartureCityNotAssociatedException,
+    InvalidArrivalTimeException,
+    InvalidDepartureTimeException,
+    SameCityException
+};
+
+class Flight extends Model
+{
+    use HasFactory;
+
+    protected static function boot()
+    {
+        parent::boot();
+
+        static::creating(function ($flight) {
+            $isDepartureCityAssociated = $flight->airline->cities()
+                ->where('city_id', $flight->departure_city_id)
+                ->exists();
+
+            if (!$isDepartureCityAssociated) {
+                throw new DepartureCityNotAssociatedException();
+            }
+
+            $isArrivalCityAssociated = $flight->airline->cities()
+                ->where('city_id', $flight->arrival_city_id)
+                ->exists();
+
+            if (!$isArrivalCityAssociated) {
+                throw new ArrivalCityNotAssociatedException();
+            }
+
+            if ($flight->arrival_city_id === $flight->departure_city_id) {
+                throw new SameCityException();
+            }
+
+            if ($flight->arrival_time <= $flight->departure_time) {
```

Comment:
Here, the time comparisions you can use the comparision methods of Carbon https://carbon.nesbot.com/docs/#api-comparison